lib/Controller/RoomController.php

```diff
@@ -1264,11 +1432,37 @@ public function joinRoom(string $token, string $password = '', bool $force = tru
 		}
 
 		$this->session->removePasswordForRoom($token);
-		$this->session->setSessionForRoom($token, $newSessionId);
-		$room->ping($this->userId, $newSessionId, $this->timeFactory->getTime());
-		$currentParticipant = $room->getParticipantBySession($newSessionId);
+		$session = $participant->getSession();
+		if ($session instanceof Session) {
+			$this->session->setSessionForRoom($token, $session->getSessionId());
+			$this->sessionService->updateLastPing($session, $this->timeFactory->getTime());
+		}
 
-		return new DataResponse($this->formatRoom($room, $currentParticipant));
+		return new DataResponse($this->formatRoom($room, $participant));
+	}
+
+	/**
+	 * @PublicPage
+	 *
+	 * @param string $pin
+	 * @return DataResponse
+	 */
+	public function getParticipantByDialInPin(string $pin): DataResponse {
```

Comment:
This also gets `string $token` which must be used to set `$this->room` in the function.

Reply:
That's done automatically with InjectionMiddleware

Reply:
For me it triggered `"Exception":"Exception","Message":"Call to a member function getToken() on null"` in `server/apps/spreed/lib/Controller/RoomController.php","Line":1452` for `/ocs/v2.php/apps/spreed/api/v2/room/5965410491/pin/2350681`. Anything I need to do so the injection works?

Reply:
Or is that related to my setup not working correctly right now?

Reply:
Fixed now